tests/ast/data/interval.test

```diff
@@ -0,0 +1,772 @@
+## TEST CASE
+
+from snowflake.snowpark._internal.analyzer.expression import Interval
+
+df1 = session.create_dataframe(
+        [
+            [datetime.datetime(2010, 1, 1), datetime.datetime(2011, 1, 1)],
+            [datetime.datetime(2012, 1, 1), datetime.datetime(2013, 1, 1)],
+        ],
+        schema=["a", "b"],
+    )
+
+df2 = df1.select(
+    df1["a"]
+    + make_interval(
+        quarters=1,
+        months=1,
+        weeks=2,
+        days=2,
+        hours=2,
+        minutes=3,
+        seconds=3,
+        milliseconds=3,
+        microseconds=4,
+        nanoseconds=4,
+    )
+)
+
+df4 = df1.select(df1["a"] + Column(Interval(1234)))
+
+df5 = df1.select(
+    df1["a"]
+    + make_interval(
+        quarters=1,
+        months=2,
+        weeks=3,
+        days=4,
+        hours=5,
+        minutes=6,
+        seconds=7
+    )
+)
+
+df6 = df1.select(
+    df1["a"]
+    + make_interval(
+        years=1,
+        months=2,
+        weeks=3,
+        days=4,
+        hours=5,
+        minutes=6,
+        seconds=7,
+        milliseconds=8,
+        microseconds=9,
+        nanoseconds=10
+    )
+)
+
```

Comment:
can you add some example where it's with
```
 make_interval(
        years=1)
``` or so only? I.e., only a few arguments set? Would be great to see that the unparser doesn't produce a lot of `None` then.

Reply:
done!